src/table.rs

```diff
@@ -24,6 +24,8 @@ const MAX_PAGES: usize = 1 << (32 - PAGE_LEN_BITS);
 
 pub struct Table {
     pub(crate) pages: boxcar::Vec<Box<dyn TablePage>>,
+    /// Map from ingredient to non-full pages that are up for grabs
+    non_full_pages: FxDashMap<IngredientIndex, Vec<PageIndex>>,
```

Comment:
I'm asking because I forgot: do we want an `FxDashMap` here, or put differently, how much contention do we expect to have on a `Table`?

Reply:
A plain map with an RwLock or Mutex is probably fine as well we would only hit this a couple times per zalsalocal